prov/sockets/src/sock_ep.c

```diff
@@ -584,7 +584,10 @@ static ssize_t sock_tx_size_left(struct fid_ep *ep)
 	fastlock_acquire(&tx_ctx->wlock);
 	num_left = rbavail(&tx_ctx->rb)/SOCK_EP_TX_ENTRY_SZ;
 	fastlock_release(&tx_ctx->wlock);
-	return num_left;
+	if (num_left > SOCK_EP_TX_SZ)
```

Comment:
> In sockets provider SOCK_EP_TX_ENTRY_SZ is not a power of two. When we allocate buffer, we roundup to the next power of two. So if fi_tx_ctx->size = 256, we allocate more than 256. So when we check the availability during fi_tx_size_left, it'll be more than 256. but since SOCK_EP_TX_SZ is the limit, I am returning the upper bound to the user.

Okay, I see what's going on now. That's a little deceiving isn't it? I see what's going on now. Maybe my test should be checking that size left returns something _greater than or equal to_ the value given by hints.